invisible_cities/cities/buffy_test.py

```diff
@@ -72,31 +72,48 @@ def test_buffy_kr(config_tmpdir, full_sim_file):
         assert np.all(sipm_integ[sipm_nonzero] == sns_sums[sns_sums.index > 12])
 
 
-def test_buffy_filters_empty(config_tmpdir, ICDATADIR):
+def test_buffy_no_file_without_sns_response(config_tmpdir, ICDATADIR):
     file_in  = os.path.join(ICDATADIR, 'nexus_new_kr83m_fast.oldformat.sim.h5')
-    file_out = os.path.join(config_tmpdir, 'filtered_empty.buffers.h5')
+    file_out = os.path.join(config_tmpdir, 'no_file.h5')
 
     nevt = 2
     conf = configure('buffy invisible_cities/config/buffy.conf'.split())
     conf.update(dict(files_in=file_in, file_out=file_out, event_range=nevt))
 
-    # Warning expected since no MC tables present.
+    try:
+        buffy(**conf)
+    except:
+        if not os.path.exists(file_out):
+            assert True
+        else:
+            assert False
+
+
+def test_buffy_filters_empty(config_tmpdir, ICDATADIR):
+    file_in  = os.path.join(ICDATADIR, 'nexus_new_kr83m_*.oldformat.sim.h5')
+    #file_in  = os.path.join(ICDATADIR, 'nexus_new_kr83m_full.oldformat.sim.h5')
+    file_out = os.path.join(config_tmpdir, 'filtered_empty.buffers.h5')
+
+    nevt     = 4
+    n_passed = 2
+    conf = configure('buffy invisible_cities/config/buffy.conf'.split())
+    conf.update(dict(files_in=file_in, file_out=file_out, event_range=nevt))
+
+    # Exception expected since no MC sensor response is present.
     # Suppress since irrelevant in test.
     with warnings.catch_warnings():
         warnings.simplefilter("ignore", category=UserWarning)
         buffy_result = buffy(**conf)
 
     assert buffy_result.events_in            == nevt
-    assert buffy_result.events_resp.n_passed == 0
+    assert buffy_result.events_resp.n_passed == n_passed
     with tb.open_file(file_out) as h5out:
-        assert h5out.root.Run.events.shape == (0,)
 
         assert hasattr(h5out.root        ,         'Filters')
         assert hasattr(h5out.root.Filters, 'detected_events')
 
         evt_filter = h5out.root.Filters.detected_events.read()
         assert len(evt_filter) == nevt
-        assert not np.any(evt_filter['passed'])
```

Comment:
Instead of removing these lines, can we put the expected values?

Reply:
Done.